src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -1166,39 +1117,73 @@ public boolean isInformative() {
         }
     }
 
-    // Requires that the collection passed iterator can remove elements, and it can be modified.
-    public void removeSampleEvidence(final int sampleIndex, final Collection<EVIDENCE> evidenceToRemove, final int alleleCount) {
-        if (evidenceToRemove.isEmpty()) {
-            return;
-        }
-
-        final List<EVIDENCE> sampleEvidence = evidenceBySampleIndex.get(sampleIndex);
-
-        final Set<EVIDENCE> evidenceToRemoveSet = new HashSet<>(evidenceToRemove);
-        final int[] evidenceIndicesToKeep = IntStream.range(0, sampleEvidence.size()).filter(r -> !evidenceToRemoveSet.contains(sampleEvidence.get(r))).toArray();
+    private void removeEvidence(final int sampleIndex, final Collection<EVIDENCE> evidences) {
+        final Object2IntMap<EVIDENCE> evidenceIndexes = evidenceIndexBySampleIndex(sampleIndex);
+        final int[] indexesToRemove = evidences.stream().mapToInt(e -> {
+            final int index = evidenceIndexes.getInt(e);
+            if (index == MISSING_INDEX) {
+                throw new IllegalArgumentException("evidence provided is not in sample");
+            } else {
+                return index;
+            }
+        }).sorted().distinct().toArray();
+        removeEvidenceByIndex(sampleIndex, indexesToRemove);
+    }
 
-        // Nothing to remove we just finish here.
-        if (evidenceIndicesToKeep.length == sampleEvidence.size()) {
+    // assumes that evidencesToRemove is sorted and without duplicates.
+    private void removeEvidenceByIndex(final int sampleIndex, final int[] evidencesToRemove) {
+        if (evidencesToRemove.length == 0) {
             return;
-        }
-
-        final List<EVIDENCE> newSampleEvidence = Arrays.stream(evidenceIndicesToKeep).mapToObj(sampleEvidence::get).collect(Collectors.toList());
-        final Object2IntMap<EVIDENCE> indexByEvidence = evidenceIndexBySampleIndex(sampleIndex);
-        indexByEvidence.clear();
-        new IndexRange(0, newSampleEvidence.size()).forEach(r -> indexByEvidence.put(newSampleEvidence.get(r), r));
-
-        // Then we skim out the likelihoods of the removed evidence.
-        final double[][] oldSampleValues = valuesBySampleIndex[sampleIndex];
-        final double[][] newSampleValues = new double[alleleCount][newSampleEvidence.size()];
-
-        for (int a = 0; a < alleleCount; a++) {
-            for (int r = 0; r < evidenceIndicesToKeep.length; r++) {
-                newSampleValues[a][r] = oldSampleValues[a][evidenceIndicesToKeep[r]];
+        } else {
+            final Object2IntMap<EVIDENCE> evidenceIndex = evidenceIndexBySampleIndex(sampleIndex);
+            final int oldEvidenceCount = numberOfEvidences[sampleIndex];
+            final int newEvidenceCount = oldEvidenceCount - evidencesToRemove.length;
+            if (newEvidenceCount < 0) {
+                throw new IllegalStateException("attempt to removed non-existant evdidence or repeated evidence");
```

Comment:
typo: evdidence

Reply:
ok, that is not the only typo in this line though.

Reply:
Indeed.